### PR TITLE
[Test] Add test to validate replica eager-lock and volume profile

### PR DESCRIPTION
**Pull request by srivickynesh**
Signed-off-by: Sri Vignesh <sselvan@redhat.com>

### CODE REVIEW

File: tests/functional/afr/test_replica_eager_lock.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            self.assertEqual(out.strip(), status_on, (
+                "Volume get failed for volume %s" % self.volname))
+
+        # Mounting a volume
+        ret = self.mount_volume(self.mounts)
+        self.assertTrue(ret, "Volume mount failed for %s" % self.volname)
+
+        # After Mounting immediately writing IO's are failing some times,
+        # that's why keeping sleep for 10 secs
+        # sleep(10)
```
Comment (pranavprakash20) [inline tests/functional/afr/test_replica_eager_lock.py:151]: Please remove if not required

File: tests/functional/afr/test_replica_eager_lock.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+#  Copyright (C) 2017-2020  Red Hat, Inc. <http://www.redhat.com>
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:1]:
```
#  Copyright (C) 2021
```
Copyright should be 2021

File: tests/functional/afr/test_replica_eager_lock.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        volume mounted successfully on client, start running IOs on
+        mount point then run the "gluster volume <volname> profile info"
+        command on all clusters randomly.
+        Then check that IOs completed successfully or not on mount point.
+        Check that files in mount point listing properly or not.
+        check the release directory value should be less than '4'
+        '''
+
+        status_on = "on"
+        validate_profiles = ["cluster.eager.lock",
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:121]: Use a tuple instead of a list

File: tests/functional/afr/test_replica_eager_lock.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+            g.log.info("IO is successful on all mounts")
+
+            # List all files and dirs created
+            g.log.info("List all files and directories:")
+            ret = list_all_files_and_dirs_mounts(self.mounts)
+            if not ret:
+                raise ExecutionError("Failed to list all files and dirs")
+            g.log.info("Listing all files and directories is successful")
+
+        volume_profile_stop = "gluster volume profile %s stop"
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:93]: Please use existing libraries

File: tests/functional/afr/test_replica_eager_lock.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    @classmethod
+    def setUpClass(cls):
+        cls.counter = 1
+        cls.get_super_method(cls, 'setUpClass')()
+
+        # checking for peer status from every node
+        ret = cls.validate_peers_are_connected()
+        if not ret:
+            raise ExecutionError("Peer probe failed ")
+        else:
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:43]: Do we really need this else?

File: tests/functional/afr/test_replica_eager_lock.py [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        """
+        # calling GlusterBaseClass setUp
+        self.get_super_method(self, 'setUp')()
+
+        # Creating Volume
+        g.log.info("Started creating volume")
+        ret = self.setup_volume()
+        if ret:
+            g.log.info("Volme created successfully : %s", self.volname)
+        else:
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:71]: Do we really need this else?

File: tests/functional/afr/test_replica_eager_lock.py [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+        ret, _, _ = g.run(random.choice(self.servers),
+                          volume_profile_stop % self.volname)
+        self.assertEqual(ret, 0, (
+            "Volume profile failed to stop for volume %s" % self.volname))
+
+        # unmounting the volume and Cleaning up the volume
+        ret = self.unmount_volume_and_cleanup_volume(self.mounts)
+        if ret:
+            g.log.info("Volume deleted successfully : %s", self.volname)
+        else:
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:103]: Do we really need this else?

**Review by kshithijiyer — CHANGES_REQUESTED** [review]

File: tests/functional/afr/test_replica_eager_lock.py [hunk 8]
```diff
@@ -0,0 +1,184 @@
+#  Copyright (C) 2017-2021 Red Hat, Inc. <http://www.redhat.com>
+#
```
Comment (kshithijiyer) [inline tests/functional/afr/test_replica_eager_lock.py:2]: I think my previous comment wasn't clear, @srivickynesh you just need `2021`.

Comment (srivickynesh) [reply tests/functional/afr/test_replica_eager_lock.py:2]: done

**Review by kshithijiyer — APPROVED** [review]
LGTM

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by balakreddy — APPROVED** [review]

**Pull request merged**